dev/source/docs/debugging-with-gdb-on-stm32.rst

```diff
@@ -65,6 +73,93 @@ for reference) to the corresponding SWDIO and SWCLK pins on the ST-Link debugger
 Ensure :ref:`BRD_OPTIONS<BRD_OPTIONS>` sets the "Enable Debug Pins" bit if bit 3 is present in the vehicle software being used.
 It is also advised to disable the watchdog by ensuring bit 0 is not set.
 
+Once BRD_OPTIONS bit 3 is set, the board will not boot until `gdb` tells it to start.
+If you try using mavproxy, it will log "no link"
+
+Connecting the ST-LINK V3 JTAG debugger to the STM32
+====================================================
+
+Install drivers. For Ubuntu x86_64:
+```
+wget https://github.com/stlink-org/stlink/releases/download/v1.7.0/stlink_1.7.0-1_amd64.deb
+sudo apt install ./stlink_1.*.deb
+rm ./stlink_1.*.deb
+```
+
+
+Connecting the Segger J-Link EDU to the STM32 on Cube Red
+=========================================================
+
+Install drivers for Segger. https://www.segger.com/downloads/jlink/
+
+Install drivers. For Ubuntu x86_64:
+```
+wget https://www.segger.com/downloads/jlink/JLink_Linux_x86_64.deb
+sudo apt install ./JLink_Linux_x86_64.deb
+rm ./JLink_Linux_x86_64.deb
+```
+
+Next, update `RTOSPlugin_ChibiOS.so`. TODO add instructions.
+Place the file in `/opt/SEGGER/JLink/GDBServer/`.
+
+
+With CubeRed wired up to both the debugger, and connected over USB-C to power it, test the connection. 
+
+For CubeRed:
+
+```bash
```

Comment:
this is causing a build error
```suggestion
::
```

Reply:
Yea, the content is really not quite there. I have some outstanding requests to CubePilot on how to get this working reliably. I really need some assistance from the dev team on this guide.